File: Java/src/com/dbc/algorithm/Math/PalindromeNumber.java
package com.dbc.algorithm.Math;

public class PalindromeNumber {
    public static boolean isPalindrome(int x) {
        if (x < 0) {
            return false;
        }
        String remain = x + "";
        int left = 0, right = remain.length() - 1;
        while (left < right) {
            if (remain.charAt(left) != remain.charAt(right)) {
                return false;
            }
            left++;
            right--;
        }
        return true;
    }
}

File: Python/math/palindrome_number.py
def isPalindrome(x: int) -> bool:
    if x < 0:
        return False
    
    remain = str(x)
    left, right = 0, len(remain) - 1
    while left < right:
        if remain[left] != remain[right]:
            return False
        left += 1
        right -= 1
    
    return True

if __name__ == "__main__":
    print(isPalindrome(x = 121))